core/store/src/trie/mod.rs

```diff
@@ -800,6 +884,94 @@ mod tests {
     type TrieChanges = Vec<(Vec<u8>, Option<Vec<u8>>)>;
     const SHARD_VERSION: u32 = 1;
 
+    impl Trie {
+        pub fn verify_proof(
+            &self,
+            key: &[u8],
+            levels: Vec<RawTrieNodeWithSize>,
+            // when verifying proofs of non-membership, this value shuold be None
+            maybe_expected_value: Option<&[u8]>,
+            mut expected_hash: CryptoHash,
+        ) -> bool {
+            let hash_node = |node: &RawTrieNodeWithSize| {
+                let mut v = vec![];
```

Comment:
what do you mean by `move` the closure?

Reply:
```rust
let mut buffer = Vec::new();
let hash_node = move |node: &RawTrieNodeWithSize| {
    node.encode_into(&mut buffer).unwrap();
    CryptoHash::hash_bytes(&buffer)
};
```
Should work.  (It might work without `move` actually).

Reply:
works without it